src/lib/core/components/visualizations/implementations/HistogramVisualization.tsx

```diff
@@ -639,3 +662,62 @@ function reorderData(
     return data;
   }
 }
+
+/**
+ * find max of the sum of multiple arrays
+ * it is because histogram viz uses "stack" option for display
+ * Also, each data with overlayVariable has different bins
+ * For this purpose, binStart is used as array index to map corresponding count
+ * Need to make stacked count array and then max
+ */
+
+function findMaxOfStackedArray(
+  // data: HistogramDataWithCoverageStatistics | undefined,
+  data: HistogramDataSeries[]
+) {
+  // construct array of objects comprised of binStart and count arrays
+  const dataArray = data.map((series) => {
+    return {
+      binStart: series.bins.map((bin) => bin.binStart),
+      count: series.bins.map((bin) => bin.count),
+    };
+  });
+
+  // find unique index and sort alphanumeric from dataArray.binStart
+  const uniqueBinStart = sortBy(
+    uniq(dataArray.flatMap((data) => data.binStart))
+  );
+
+  // make array of count array (2D)
+  let totalCountArray = [];
+
+  // construct 2D totalCountArray
+  // [# of stratification] rows and [# of unique binStart] at each low
+  if (uniqueBinStart != null) {
+    for (let i = 0; i < dataArray.length; i++) {
```

Comment:
A [for...of](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/for...ofl) loop might improve readability.